src/elements/frame-router.ts

```diff
@@ -102,6 +102,8 @@ export default class FrameRouterElement extends HTMLElement {
    * @param newPath a new route which matches those provided originally.
    */
   public changeRoute(newPath: string) {
+    newPath = newPath || 'about:blank';
```

Comment:
This doesn't really make sense, because we don't want to `getClientTarget` on `about:blank`.  As currently written, the idea is that if `newPath` is invalid in any way `clientInfo` should be null, and `setFrameLocation` on line 117/119 should be called with `null` and then `setFrameLocation` sets `about:blank` ([see here](https://github.com/purecloudlabs/iframe-coordinator/blob/master/src/FrameManager.ts#L85)).  

I think if we are trying to fall-back to `about:blank` in multiple locations, it makes things harder to understand, and opens us up to more potential bugs in the future. Can you figure out why the current mechanism isn't working?

Reply:
The current mechanism isn't working because we expect the passed value to be a string and not null / undefined / etc and our string processing tools break trying to use string methods (`matchAndStripPrefix` being the primary offender).

An alternative fix would be to return the "not found" case from client target lookup if no route was passed
```
let clientTarget: ClientTarget = {
      id: null,
      url: null,
      assignedRoute: null
};
```

Then the frame router (here) would behave as if no viable client was found and default to about:blank without any further changes. Would that be reasonable?

Reply:
Looking at the code a bit more, I think this may stem from bad type signatures in `attributeChangedCallback` where we don't account for attributes possibly being null. We should update that type signature accordingly and follow it through.   

That generally means making the changes you suggest but instead of having `getClientTarget` return an object full of nulls when no client is found, we should probably just return null. That way we avoid the weird invalid state ambiguity of what it means if some properties are null and others are not.